server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodeFilters.java

```diff
@@ -88,6 +89,34 @@ private boolean matchByIP(String[] values, @Nullable String hostIp, @Nullable St
         return false;
     }
 
+    /**
+     * Removes any filters that should not be considered, returning a new
+     * {@link DiscoveryNodeFilters} object. If the filtered object has no
+     * filters after trimming, {@code null} is returned.
+     */
+    @Nullable
+    public static DiscoveryNodeFilters trim(@Nullable DiscoveryNodeFilters original) {
```

Comment:
nit: maybe we should specify what we are trimming in the method name? (so the `FilterAllocationDecider` code is a bit more readbale without having to step into the `trim` method to see what exactly is removing). Maybe `trimTierFilters` or something similar?

Reply:
Sure, I can rename that to make it clearer